packages/icons/src/metadata.ts

```diff
@@ -0,0 +1,1816 @@
+  },
+  "user-nametag": {
+    category: "People & User",
+  },
+  "user-plus": {
+    category: "People & User",
+  },
+  "user-settings": {
+    category: "People & User",
+  },
+  "user-shield": {
+    category: "People & User",
+  },
+  "user-suit-nametag": {
+    category: "People & User",
+  },
+  "user-x": {
+    category: "People & User",
+  },
+  vibration: {
+    category: "Communication",
+  },
+  videocam: {
+    category: "Devices & Electronics",
+  },
+  "videocam-disabled": {
+    category: "Devices & Electronics",
+  },
+  virus: {
+    category: "Health, Medicine & Hygiene",
+  },
+  "virus-shield": {
+    category: "Health, Medicine & Hygiene",
+  },
+  "volume-disabled": {
+    category: "Action & Interface",
+  },
+  "volume-down": {
+    category: "Action & Interface",
+  },
+  "volume-mute": {
+    category: "Action & Interface",
+  },
+  "volume-up": {
+    category: "Action & Interface",
+  },
+  voucher: {
+    category: "Commerce & Shopping",
+  },
+  walking: {
+    category: "Transport & Travel",
+  },
+  wallet: {
+    category: "Commerce & Shopping",
+  },
+  warehouse: {
+    category: "Commerce & Shopping",
+  },
+  webhook: {
+    category: "Software & Services",
+  },
+  windmill: {
+    category: "Weather & Climate",
+  },
+  windows: {
+    category: "Software & Services",
+  },
+  wlan: {
+    category: "Connectivity",
+  },
+  wrench: {
+    category: "Objects",
+  },
+  x: {
+    category: "Essentials",
+  },
+  "x-small": {
+    category: "Essentials",
+  },
+  "zoom-in": {
+    category: "Action & Interface",
+  },
+  "zoom-out": {
+    category: "Action & Interface",
+  },
+} as const satisfies Record<string, IconMetadata>;
+
+/**
+ * Grouped metadata of all available icons by category.
+ */
+export const ICON_CATEGORIES = groupIconsByCategory();
+
+export type IconMetadata = {
+  category: string;
+  deprecated?: boolean;
+  aliases?: string[];
+};
+
+export type GroupedIconCategory = {
+  iconName: keyof typeof ICON_METADATA;
+  metadata: IconMetadata;
+};
+
+export type IconCategories = Record<string, GroupedIconCategory[]>;
+
+/**
+ * Groups all available icon metadata by category.
+ * Categories and icons will be sorted alphabetically.
+ */
+function groupIconsByCategory() {
+  const categories = Object.entries(ICON_METADATA).reduce<IconCategories>(
+    (categories, [iconName, metadata]) => {
+      const icons = categories[metadata.category] ?? [];
+      icons.push({ iconName: iconName as GroupedIconCategory["iconName"], metadata });
+      categories[metadata.category] = icons;
+      return categories;
+    },
+    {},
+  );
+
+  const sortedCategories: typeof categories = {};
+
+  Object.keys(categories)
+    .sort()
+    .forEach((category) => {
+      const sortedMetadata = categories[category].slice().sort((a, b) => {
+        return a.iconName.localeCompare(b.iconName);
+      });
+      sortedCategories[category] = sortedMetadata;
+    });
+
+  return sortedCategories;
```

Comment:
in lack of a unit test I can not be sure, but shouldn't this do the same thing?

```suggestion
  return Object.entries(ICON_METADATA).reduce<IconCategories>(
    (categories, [iconName, metadata]) => {
      const icons = categories[metadata.category] ?? [];
      icons.push({ iconName: iconName as GroupedIconCategory["iconName"], metadata });
      icons.sort((a, b) => {
        return a.iconName.localeCompare(b.iconName);
      });
      categories[metadata.category] = icons;
      return categories;
    },
    {},
  );
```

Reply:
here's the better readable diff for you :)

<img width="1300" alt="image" src="https://github.com/SchwarzIT/onyx/assets/151019977/492cf7c3-0789-4589-979f-930bc33d65d6">

Reply:
This only sorts the icons, not the categories itself. Also its less performant since all icons are sorted in every iteration instead of once after all icons are grouped.

Reply:
ok